### PR TITLE
DOV-6868: Port config rewrite - Part 2

### PR DESCRIPTION
**Pull request by fleischie**
Another batch of changes.

Requires:
- #952 
- #953 

Closes DOV-6868 (eventually).

### CODE REVIEW

File: docs/core/config/statistics.md [hunk 1]
```diff
@@ -20,44 +20,19 @@ Dovecot supports gathering statistics from events (see [[link,event_design]]).
 Currently there are no statistics logged by default, and therefore they must
 be explicitly added using the `metric` configuration blocks.
 
-The event filter settings are the only required settings in a metric block.
-The filter specifies which events should be used when calculating the
-statistics for a given metric block.  Event filtering is described in detail
-in [[link,event_filter]].
+The event `filter` setting are the only required settings in a `metric` block.
```
Comment (sirainen) [inline docs/core/config/statistics.md:23]: * filter can be a link?
 * are .. settings -> is .. setting
 * metric can be a link?

File: data/settings.js [hunk 2]
```diff
@@ -4990,6 +5044,88 @@ It can include key = value pairs for assigning variables the desired value
...
+	inet_listener_port: {
+		tags: [ 'service' ],
+		values: setting_types.UINT,
+		default: 0,
+		text: `
+Port number where to listen. \`0\` disables the listener.
+
+::: warning
+Although this is an unsigned integer. Only values between \`0\` and \`65535\`
+are supported.
```
Comment (sirainen) [inline data/settings.js:5049]: Also complained a bit about this to Michael's intercept merge request that we really should have setting_types.IN_PORT like in the code there is already. Then you wouldn't need to sprinkle around these kind of warnings.

Also, it should be a single sentence. Probably with comma: `integer, only`

File: docs/core/config/service.md [hunk 3]
```diff
@@ -681,19 +430,17 @@ logging via syslog.
...
 
   Dovecot attempts to verify that the limit is high enough at startup.
   If it's not, it logs a warning such as:
 
-    * Warning: service stats { client_limit=1000 } is lower than required under max. load (7945)
+  `Warning: service stats { client_limit=1000 } is lower than required under max. load (7945)`
 
-      This is calculated by counting the `process_limit` of all the services,
-      because each of them has a persistent connection to stats.
+  This is calculated by counting the `process_limit` of all the services,
```
Comment (sirainen) [inline docs/core/config/service.md:445]: process_limit link?

**Review by sirainen — CHANGES_REQUESTED** [review]

File: docs/core/config/statistics.md (hunk 1 above)
Comment (fleischie) [reply docs/core/config/statistics.md:23]: Where do the links point to, though?

File: data/settings.js (hunk 2 above)
Comment (fleischie) [reply data/settings.js]: I added a new type and replaced the current values for all available port settings as of this MR.

File: docs/core/config/statistics.md (hunk 1 above)
Comment (sirainen) [reply docs/core/config/statistics.md]: metric_filter and metric settings? Not sure if they exist in this pull request yet.

File: data/settings.js (hunk 2 above)
Comment (sirainen) [reply data/settings.js]: I don't see any such changes. Did you push them?

Comment (sirainen) [reply data/settings.js]: Hmm no, I'm just having trouble with github

Comment (fleischie) [reply data/settings.js]: No, not yet. I will keep an eye out for these beginning now.

**Review by sirainen — APPROVED** [review]

**Pull request merged**